public/js/test/mochitest/browser_dbg-navigation.js

```diff
@@ -0,0 +1,30 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+function countSources(dbg) {
+  const sources = dbg.selectors.getSources(dbg.getState());
+  return sources.size;
+}
+
+/**
+ * Test navigating
+ * navigating while paused will reset the pause state and sources
+ */
+add_task(function* () {
+  const dbg = yield initDebugger(
+    "doc-script-switching.html",
+    "script-switching-01.js"
+  );
+
+  invokeInTab("firstCall");
+  yield waitForPaused(dbg);
+
+  yield navigate(dbg, "doc-iframes.html");
+  yield waitForPaused(dbg);
+  assertPausedLocation(dbg, "iframes.html", 8);
+  ok(countSources(dbg) == 1);
+
+  yield navigate(dbg, "about:blank");
+  yield waitForDispatch(dbg, "NAVIGATE");
+  ok(countSources(dbg) == 0);
+});
```

Comment:
I'm sure you're planning on expanding this test, but you could go ahead and navigate back to `doc-scripts.html` and assert that there are 3 sources again.